From e7effe90f4f9d88f58bca975c76ebdac8e161c92 Mon Sep 17 00:00:00 2001
From: Shawn Ferry <sferry@gmail.com>
Date: Thu, 11 May 2017 15:12:53 -0400
Subject: [PATCH] 26045611 service name needs to allow colon (:)

26045611 mcollective service name needs to allow colon (:)

Patch is accepted by upstream
(MCOP-594) service plugin unable to manage services with ':' in the name

---
 validator/service_name.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/validator/service_name.rb b/validator/service_name.rb
index a9c450a..7e07e67 100644
--- a/validator/service_name.rb
+++ b/validator/service_name.rb
@@ -2,7 +2,7 @@ module MCollective
   module Validator
     class Service_nameValidator
       def self.validate(service_name)
-        raise("%s is not a valid service name" % service_name) unless !!(service_name =~ /\A^[a-zA-Z\.\-_\d@]+$\z/)
+        raise("%s is not a valid service name" % service_name) unless !!(service_name =~ /\A^[a-zA-Z\.\-_\d:@]+$\z/)
       end
     end
   end
-- 
2.11.0 (Apple Git-81)

